src/DynamoPackages/PackageLoader.cs

```diff
@@ -375,13 +375,16 @@ public void LoadPackages(IEnumerable<Package> packages)
                 }
             }
         }
+
         /// <summary>
-        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
+        /// Helper function to load new custom nodes and packages.
         /// </summary>
-        /// <param name="newPaths"></param>
-        /// <param name="preferences"></param>
+        /// <param name="preferences">Can be a temporary local preferences object.</param>
         /// <param name="customNodeManager"></param>
-        public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreferences preferences, CustomNodeManager customNodeManager)
+        /// <param name="newPaths">New package paths to load custom nodes and packages from.</param>
+        /// <param name="packageDirectoriesToScan">Scan new paths if this is null.</param>
+        private void LoadCustomNodesAndPackagesHelper(IPreferences preferences, 
+            CustomNodeManager customNodeManager, IEnumerable<string> newPaths, IEnumerable<string> packageDirectoriesToScan = null)
```

Comment:
can;t you use a single `IEnumerable<string> searchPaths` parameter instead of the 2 `IEnumerable<string> newPaths, IEnumerable<string> packageDirectoriesToScan = null` ?

Reply:
They have different meanings in the context of the internal method.

Reply:
Ok, will check.